packages/airnode-utilities/src/caching/index.ts

```diff
@@ -0,0 +1,252 @@
+import { existsSync, mkdirSync, readdirSync, readFileSync, rmdirSync, rmSync, statSync, writeFileSync } from 'fs';
+import { spawn } from 'child_process';
+import { join } from 'path';
+import { logger } from '../logging';
+
+export const CACHE_BASE_PATH = `/tmp/airnode-cache`;
+export const CACHE_MAX_FILESYSTEM_AGE_MINUTES = 60;
+export const CACHE_MAX_FILES = 100_000; // (500 (cache size) * 1024 (KB) ) / 4 (KB, fs "cluster size" = 130k
+
+export const isJest = () => process.env.JEST_WORKER_ID !== undefined;
+
+/**
+ * Calls the OS sync command to forcefully sync the cache filesystem.
+ * This is the functional equivalent of "commit" from SQL.
+ */
+const syncFsSync = () => {
+  try {
+    // TODO find a way to mock out spawn...
+    if (!isJest()) {
+      spawn(`sync`, [`-f`, CACHE_BASE_PATH]);
+    }
+  } catch (e) {
+    logger.error(`Unable to sync cache fs`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Syncs the temporary filesystem asynchronously.
+ * Lambda will "freeze" and "thaw" the execution context and with it any async promises in progress. We create and
+ * forget this promise here, but it will resolve at some point, either in this invocation or a later one.
```

Comment:
After some more thought calling `sync` is probably unnecessary;
file system access should be thread safe and the cache is intended to be used in an append-only way (we create files, we don't modify them).
Explicitly calling `sync` is often used by databases to commit transactions to disk so that the database can be recovered in the event of a system crash or power outage, but this is not a problem here because we can expect cloud to be reliable and we don't care about the longevity of the cache beyond a few minutes.
So I think I'll remove it entirely.

Reply:
Yeah, that makes even more sense to me.